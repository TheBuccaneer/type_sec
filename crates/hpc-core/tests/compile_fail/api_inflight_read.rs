use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();

    // Buffer → Ready
    let buf = ctx
        .create_buffer::<u8>(16).unwrap()
        .enqueue_write(&queue, &[0u8; 16]).unwrap();

    // Kernel starten → InFlight + Event
    let (inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();

    let mut out = [0u8; 16];

    // Verboten: Read auf InFlight
    inflight.enqueue_read_blocking(&queue, &mut out).unwrap();
}
