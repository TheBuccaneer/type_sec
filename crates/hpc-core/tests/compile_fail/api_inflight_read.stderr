warning: unreachable expression
  --> tests/compile_fail/api_inflight_read.rs:13:55
   |
13 |     let (inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();
   |                                                       ^----------------
   |                                                       ||
   |                                                       |any code following this expression is unreachable
   |                                                       unreachable expression
   |
   = note: `#[warn(unreachable_code)]` on by default

error[E0308]: mismatched types
  --> tests/compile_fail/api_inflight_read.rs:13:55
   |
13 |     let (inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();
   |                                --------------         ^^^^^^^^^^^^^^^^^ expected `&Kernel<'_>`, found `&!`
   |                                |
   |                                arguments to this method are incorrect
   |
   = note: expected reference `&Kernel<'_>`
              found reference `&!`
note: method defined here
  --> src/api/device_buffer/ready/mod.rs
   |
   |     pub fn enqueue_kernel(
   |            ^^^^^^^^^^^^^^

error[E0599]: no method named `enqueue_read_blocking` found for struct `DeviceBuffer<'_, u8, InFlight>` in the current scope
  --> tests/compile_fail/api_inflight_read.rs:18:14
   |
18 |     inflight.enqueue_read_blocking(&queue, &mut out).unwrap();
   |              ^^^^^^^^^^^^^^^^^^^^^ method not found in `DeviceBuffer<'_, u8, InFlight>`
   |
   = note: the method was found for
           - `DeviceBuffer<'brand, T, hpc_core::Ready>`
